plugins/mongodb/README.md

```diff
@@ -0,0 +1,248 @@
+```
+cd ./mongodb-build
+cmake -DMONGODBEMBED=ON ../HPCC-Platform
+make -j3 package
+sudo dpkg -i ./hpccsystems-plugin-mongodbembed_<version>.deb
+```
+
+## Documentation
+
+[Doxygen](https://www.doxygen.nl/index.html) can be used to create nice HTML documentation for the code. Call/caller graphs are also generated for functions if you have [dot](https://www.graphviz.org/download/) installed and available on your path.
+
+Assuming `doxygen` is on your path, you can build the documentation via:
+```
+cd plugins/mongodb
+doxygen Doxyfile
+```
+
+The Documentation can then be accessed via `mongodb/docs/html/index.html`.
+
+## Usage
+
+To start using the plugin in your ECL code you must import mongodb before you create the EMBED statement.
+```
+IMPORT MongoDB;
+```
+
+You then need to define a function for interfacing with MongoDB. Within the EMBED statement is where you can write code to interact with the MongoDB database or collection that you have passed into the EMBED options. The plugin only supports one line and one operation per EMBED statement. If you wish to chain multiple operations together the aggregate function is supported and can take an aggregation pipeline as long as it is valid BSON.
+
+### Options
+
+To create the uri for the MongoDB connection instance the ECL user needs to pass in the username, password, server name, or just the port to use for connecting to the cluster. The plugin takes all of these and creates a shared connection instance for many threads to have access to the MongoDB databases. The plugin can only connect to one MongoDB cluster per workunit, but it can access multiple databases and collections if multiple functions are defined. The ECL user can define the batch size of the result rows.
+
+| Option | Description |
+| ------ | ----------- |
+| user   | Username of user with read and write priveleges to MongoDB server. |
+| password | Password of user with read and write priveleges to MongoDB server. |
+| server | Server connection string for connecting to MongoDB Atlas. (cluster0.qdvfhrk.mongodb.net) |
+| port | Port number for connecting to a local MongoDB server. |
+| database | Name of the database to issue commands to. |
+| collection | Name of the collection to issue commands to. |
+| batchSize | Batch size of cursor to result records. The default batch size is 100 meaning the cursor fetches 100 documents at a time from MongoDB. |
+
+```
+integer getCount() := EMBED(mongodb : user(user), password(pwd), server(server), database(dbname),  collection(collname), batchSize(100))
+ENDEMBED; 
+```
+
+For connecting to a local MongoDB instance you just need to pass in the port number that the server is listening on.
+
+```
+integer getCount() := EMBED(mongodb : port(port), database(dbname),  collection(collname))
+ENDEMBED; 
+```
+
+### Parameters
+
+To pass paremeters of the function you are defining into the MongoDB query insert them anywhere in the BSON document with a leading '$'.
```

Comment:
Nit-pick:  I think the wording here is "to *use* function parameters within the MongoDB statement, prefix them with a '$'" or something along those lines.  You're passing them just fine now, without the $.